src/helpers.ts

```diff
@@ -56,3 +57,222 @@ export function removeOrSaveUser(user: User): void {
     }
   }
 }
+
+export function weightedPoolSwap(
+  event: ethereum.Event,
+  id: string,
+  gnoIn: BigInt,
+  gnoOut: BigInt
+): void {
+  const pool = WeightedPool.load(id);
+  if (!pool) {
+    log.warning(
+      "Weighted pool with id {} could not be loaded. Trying to handle {}#{}",
+      [
+        id,
+        event.transaction.hash.toHexString(),
+        event.transactionLogIndex.toString(),
+      ]
+    );
+    return;
+  }
+
+  // Swap() is emitted after Sync(), so the balance should be up to date
```

Comment:
How does the situation look for balancer? Is Swap emitted after pool.gnoBalance has been updated already?

Reply:
It seems like we don't update `gnoBalance` for Balancer V1 anywhere else, so this won't quite work. It's rather:

```
const gnoReservesBefore = pool.gnoBalance;
const gnoReserves = gnoReserves.plus(gnoIn).minus(gnoOut);
```

Reply:
shouldn't we also be able to use this for Balancer V2 now?

Reply:
I think so. Let me push some changes, and then i adapt it for balancerv2 to use the helper as well

Reply:
The comment is a good idea.

I think yes we can also use the helper for balanceV2. Will push